hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -688,14 +688,17 @@ void sawNoSuchFamily() {
   // Last flush time for each Store. Useful when we are flushing for each column
   private final ConcurrentMap<HStore, Long> lastStoreFlushTimeMap = new ConcurrentHashMap<>();
 
-  final RegionServerServices rsServices;
+  protected RegionServerServices rsServices;
```

Comment:
Similar to how we passed `conf` all the way from HBaseTestingUtility methods to actual `HRegion` instance, we can also pass `rsServices` instance in same methods and we won't have to worry about changing this to non-final, but not a strong opinion if this means too many args in testing utility methods so it's upto you.

Reply:
In the HRegion tests I mock RSS with mockito, to test if the HRegion code calls RSS#abort as it should. We don't need to make a real cluster test that aborts a server (and adds a lot of time to the unit test suite) just to confirm this. So, it needs to be nonfinal for mocking, but it does not need to be exposed as a settable thing, because nowhere else in the code is there a use for that.